docs/hash-map.rst

```diff
@@ -0,0 +1,578 @@
+performed and the values of the map are replaced with the results from the
+function. You can think of this as a ``map`` with affects.
+
+::
+
+    -- | Ask the user how they want to schedule a bunch of tasks
+    -- that the boss has assigned certain priorities.
+    makeSchedule :: HashMap Task Priority -> IO (HashMap Task DateTime)
+    makeSchedule = traverseWithKey $ \task priority ->
+      do
+        putStrLn $ "The boss thinks " ++ show task ++
+	             " has priority " ++ show priority ++
+                     ". When do you want to do it?"
+        readLn
+
+
+
+Set-like Operations
+^^^^^^^^^^^^^^^^^^^
+
+.. _union:
+
+Union
+"""""
+
+::
+
+    HashMap.unionWith :: (v -> v -> v) -> HashMap k v -> HashMap k v -> HashMap k v
+    HashMap.unionWith f l r = ...
+
+:haddock_short:`/Data.HashMap.Strict#union` returns a map containing all entries that
+are keyed in either of the two maps. If the same key appears in both maps, the
+value is determined by calling ``f`` passing in the left and right value (`set
+union <https://en.wikipedia.org/wiki/Union_(set_theory)>`_).
+
+::
+
+
+    HashMap.unionWith (++) HashMap.empty (HashMap.fromList [(1,"x"),(2,"y")])
+    > fromList [(1,"x"),(2,"y")]
+
+    let f lv rv = lv
+    HashMap.unionWith f (HashMap.fromList [(1, "a")]) (HashMap.fromList [(1,"x"),(2,"y")])
+    > fromList [(1,"a"),(2,"y")]
+
+    HashMap.unionWith (++) (HashMap.fromList [(1, "a")]) (HashMap.fromList [(1,"x"),(2,"y")])
+    > fromList [(1,"ax"),(2,"y")]
+
+
+Intersection
+""""""""""""
+
+::
+
+    HashMap.intersectionWith :: (v -> v -> v) -> HashMap k v -> HashMap k v -> HashMap k v
+    HashMap.intersectionWith f l r = ...
+
+:haddock_short:`/Data.HashMap.Strict#intersection` returns a map containing all
+entries that have a key in both maps ``l`` and ``r``. The value in the returned
+map is determined by calling ``f`` on the values from the left and right map
+(`set intersection <https://en.wikipedia.org/wiki/Intersection_(set_theory)>`_).
+
+::
+
+    HashMap.intersectionWith (++) HashMap.empty (HashMap.fromList [(1,"x"), (2,"y")])
+    > fromList []
+
+    HashMap.intersectionWith (++) (HashMap.fromList [(1, "a")]) (HashMap.fromList [(1,"x"),(2,"y")])
+    > fromList [(1,"ax")]
+
+
+
+Difference
+""""""""""
+
+::
+
+    HashMap.difference :: HashMap k v -> HashMap k v -> HashMap k v
+    HashMap.difference l r = ...
+
+:haddock_short:`/Data.HashMap.Strict#difference` returns a map containing all
+entries that have a key in the ``l`` map but not the ``r`` map (`set
+difference/relative complement
+<https://en.wikipedia.org/wiki/Complement_(set_theory)#Relative_complement>`_).
+
+::
+
+    HashMap.difference (HashMap.fromList [(1,"one"), (2,"two"), (3,"three")]) HashMap.empty
+    > fromList [(1,"uno"),(2,"two"),(3,"three")]
+
+    HashMap.difference (HashMap.fromList[(1,"one"), (2,"two")]) (HashMap.fromList [(1,"uno")])
+    > fromList [(2,"two")]
+
+
+Serialization
+-------------
+
+TODO(m-renaud): Serialization docs.
```

Comment:
This is rendered in the [docs](https://m-renaud-haskell-unordered-containers.readthedocs.io/en/m-renaud-docs-user-guide/hash-set.html#serialization). Wrong format, or still a valid TODO?

Reply:
Valid Todo, I was going to ask if there is a recommended way to serialize the unordered-container data types. Many serialization libraries support containers (https://haskell-containers.readthedocs.io/en/latest/set.html#serialization for example) but I wasn't sure what's best to put here. Maybe just omit this section for now (here and where your other comment is)? Open to suggestions :)

Reply:
I think that's a question for the expert - @treeowl ^

Reply:
Removed Serialization sections for the time being, can add them in a follow-up PR if desired.